components/match2/wikis/ageofempires/opponent_custom.lua

```diff
@@ -0,0 +1,33 @@
+---
+-- @Liquipedia
+-- wiki=ageofempires
+-- page=Module:Opponent/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Table = require('Module:Table')
+local Opponent = require('Module:Opponent')
+local Abbreviation = require('Module:Abbreviation')
+
+
+local CustomOpponent = Table.deepCopy(Opponent)
+
+function CustomOpponent.resolve(opponent, date, options)
+	Opponent.resolve(opponent, date, options)
+	if Opponent.typeIsParty(opponent.type) then
+		local PlayerExt = require('Module:Player/Ext/Custom')
+		for _, player in ipairs(opponent.players) do
+			if Opponent.playerIsTbd(player) then
+				player.displayName = Abbreviation.make('TBD', 'To be determined (or to be decided)')
+			else
+				if not player.team then
+					player.team = PlayerExt.syncTeam(player.pageName, nil)
+				end
+			end
```

Comment:
1) why else and then if instead of `elseif`?
2) i don't think you always want that, e.g. in matches (match2 uses this function too iirc) --> imo lock it behind `options.syncPlayer` aswell

```suggestion
			elseif not player.team and options.syncPlayer then
				player.team = PlayerExt.syncTeam(player.pageName, nil)
			end
```